### DISCUSSION TITLE
Migrating from an old 0.45 build to latest on Arm

### DISCUSSION THREAD

**Discussion by AMoonRabbit** [TriliumNext/Trilium#3565] [Q&A] [answered]
Hi all,

I am looking to update my trillium instance (docker on pi4B so ARM based) but so far have come unstuck with the gap between old and current.

I note that zadam's repo only started supporting ARM arch from 0.48, but when I have tried to build and restore my database on the current release (0.5X), the logs report that the database must have been used with v0.47 at the minimum. Well, hlince's repo doesn't go that far, and with no ARM based version 0.47 for the zadam releases what are my options?

tl;dr: Currently using 0.45 on ARM, cannot migrate to current due to database incompatability.

**Comment by zadam** [comment, marked as answer]
I think the easiest would be to export the whole content into ZIP (right click in the tree -> export, HTML as ZIP).

Then run a new version and import that ZIP. This should preserve everything except for note revisions.

Another way would be to copy the document.db into a computer where you can migrate to 0.47, do the migration there and upload back to server.

**Comment by AMoonRabbit** [comment]
Thank you for answering, I went down the route of running the desktop version of 47.2 on my Windows PC, and used my documents.db file there. And then copied it back to the update instance and it's successfully migrated.
